components/role-mgt/org.wso2.carbon.identity.role.mgt.core/src/main/java/org/wso2/carbon/identity/role/mgt/core/dao/RoleDAOImpl.java

```diff
@@ -575,6 +578,39 @@ public RoleBasicInfo updateUserListOfRole(String roleID, List<String> newUserIDL
         return new RoleBasicInfo(roleID, roleName);
     }
 
+    private void validateUserRemovalFromRole(List<String> deletedUserNamesList, String roleName, String tenantDomain)
+            throws IdentityRoleManagementException {
+
+        if (!IdentityUtil.isSystemRolesEnabled() || deletedUserNamesList.isEmpty()) {
+            return;
+        }
+        try {
+            String fullyQualifiedUsername = CarbonContext.getThreadLocalCarbonContext().getUsername();
+            String username = UserCoreUtil.removeDomainFromName(fullyQualifiedUsername);
+            String adminUserName = CarbonContext.getThreadLocalCarbonContext().getUserRealm().getRealmConfiguration()
+                    .getAdminUserName();
+            // Only the tenant owner can remove users from Administrator role.
+            if (RoleConstants.ADMINISTRATOR.equalsIgnoreCase(roleName)) {
+                if (!StringUtils.equals(username, adminUserName)) {
```

Comment:
check equals or equalIgnoreCase based on the caseInsensitiveUsernameIncache property